rest/blip_sync.go

```diff
@@ -759,7 +759,6 @@ func (bh *blipHandler) sendRevisionWithProperties(body db.Body, sender *blip.Sen
 			defer func() {
 				if panicked := recover(); panicked != nil {
 					base.Warnf(base.KeyAll, "[%s] PANIC handling 'sendRevision' response: %v\n%s", bh.blipContext.ID, panicked, debug.Stack())
-					bh.close()
```

Comment:
What happens in scenarios where this would have been the first error (as opposed to the attachment/timeout  case)?  Would it be preferable to modify close() to include 'once'-style handling?

Reply:
The blipSyncContext is _always_ closed from inside `handleBLIPSync`, when `server.ServeHTTP(h.response, h.rq)` unblocks, we shouldn't be calling this from anywhere else.

https://github.com/couchbase/sync_gateway/blob/fd30c86bc3147446ef3abdcb94f42079f961c4ed/rest/blip_sync.go#L127-L133

I think having an "only-once" type thing in close is only going to encourage incorrect usages of it.

Reply:
That's understood - I was asking about the case where the connection is still live (server.ServeHTTP doesn't unblock), but there's a panic in outrq.Response().  It looks like there are a few scenarios that would result in a panic (request.Type() != requestType, request.number==0) - is it appropriate to leave the handler open in those cases?